### DISCUSSION TITLE
Dynamic hide submit button

### DISCUSSION THREAD

**Discussion by darren-glanville** [filamentphp/filament#8198] [Help] [answered]
How do you go about dynamically hiding the submit button from a resource create and edit form.

I want to disable / hide the submit button if a user has not not met all the requirements.

Thanks in advance

**Comment by leandrocfe** [comment]
You can add `disabled` or `visible` methods to the create button in the CreatePage, like this

```php
protected function getCreateFormAction(): \Filament\Actions\Action
{
    return parent::getCreateFormAction()
        ->disabled(function (): bool {
            //condition here...
        });
}
```

**Reply by darren-glanville** [reply]
Thanks, can you hide it depending on a Closure such as $get on a value of a field?

**Reply by danharrin** [reply, marked as answer]
`$this->data['field_name']`

**Reply by darren-glanville** [reply]
Great thanks that worked

**Reply by rasul-chopurov** [reply]
how to disable Prevent form submit on Enter  ?

**Reply by saidqb** [reply]
if you want dynamic hide submit button, from action from tab change,
```
Actions\Action::make($action_status)

->extraAttributes([
    'x-init' => "
    window.addEventListener(`DOMContentLoaded`, function(event) {
    qtab = new URLSearchParams(location.search).get(`tab`);
    if(typeof qtab != `undefined`){
        if(qtab == `ort-laman-2-tab`){
            show = true;
        }
    }
    });
    window.addEventListener(`click`, function(){
    qtab = new URLSearchParams(location.search).get(`tab`);
    show = false;
        if(qtab == `ort-laman-2-tab`){
            show = true;
        }
    });
    ",
    'x-data' => "{ show: false }",
    'x-show' => "show",
])
```
more hack form filament v3, from my experience
[my post](https://saidqb.github.io/filamentphp-v3-form/)

**Comment by stblink** [comment]
Hey @darren-glanville , 

Wondering how did you achieve this? Using the method above i can hide the button based on condition but that is only when i first enter the resource edit form. So there is no reactiveness. Lets say i have a toggle button in the edit resource form, if its false the create button should be hidden, but that should happen as soon as i press the toggle button and not on page refresh.

thanks in advance

**Comment by darren-glanville** [comment]
Hello, this should work for you, I have removed the bits specific to the site and tried to make it generic:

**Within a Resource and the Pages folder** - edit the Create or Edit page (for instance CreateBooking.php page)

```php
<?php

namespace App\Filament\Resources\BookingResource\Pages;

use App\Filament\Resources\BookingResource;

use Filament\Resources\Pages\CreateRecord;
use Illuminate\Support\Facades\Auth;

class CreateBooking extends CreateRecord
{
    protected static string $resource = BookingResource::class;

    protected function getFormActions(): array
    {
        $data = $this->data; // you can use the data directly from the form

        if (auth()->user()->canBook() && $this->data['agreed'] == true) { // can the user book and have they checked the agreed checkbox within the form
            return [
                $this->getCreateFormAction(),
                $this->getCancelFormAction()
            ];
        } else {
            return [];
        }
    }
```

You can also apply the above to the Edit page (ie EditBooking.php) with the added bonus of being able to use `$this->record->name` to get what the name of the record already is.

Hope that helps?

**Reply by stblink** [reply]
Thanks, it helped it me solve the problem!

**Comment by sairiz** [comment]
Hi, i am looking for a way to disable/hide the create button on the form when using CreateAction. 

getFormActions() only work for the normal form on resource page but i create a modal form using CreateAction it doesnt work

**Reply by darren-glanville** [reply]
Hello @sairiz  

Sorry for the delay.

Just had a play in Filament and was able to get the submit button being shown or hidden based on the state of the livewire mounted action data:

```php
Action::make('testing')
    ->form([
        // show_submit field
        Toggle::make('show_submit')
            ->live()
    ])
    ->modalFooterActions(
        function ($livewire, $action) {
            // get mounted data
            $data = collect($livewire->mountedTableActionsData[0]);

            // get state of show_submit
            $show_submit = $data->get('show_submit', false);

            // show buttons based on state
            return $show_submit ? [$action->getModalSubmitAction(), $action->getModalCancelAction()] : [$action->getModalCancelAction()];
        }
    )
```

It should work for the CreateAction too as uses the same action bits from Filament.

Give me a shout if that works or not.

**Reply by sairiz** [reply]
HI, thanks for the tip. I too play with modalFooterAction until i stumble upon direct way to manipulate the submit button.

```
->modalSubmitAction(function (StaticAction $action, $livewire) {
                        $action->color(fn (): string => $livewire->kodAvailability ? 'success' : 'gray');
                        $action->disabled(fn (): bool => ! $livewire->kodAvailability !== false);
                        $action->label(new HtmlString(Blade::render('<span class="inline-flex items-center"><span class="mr-1 align-middle">Create</span><x-untitledui-code-snippet class="w-5 h-5" /></span>')));
                    })
```

I guess to hide/show we would use visible/hidden method. Havent tried that though.

We could actually use the static action as a parameter to directly modify the button to our liking. Hopefully its also useful for somebody else who are looking for similar solutions.

Thanks!
